src/firefly/js/tables/FilterInfo.js

```diff
@@ -22,14 +25,30 @@ export const FILTER_TTIPS =
 * string value must be enclosed in single quotes
 * when IN is used, enclose the values in parentheses
 * you may combine conditions with either 'and' or 'or'
-* grouping by parentheses is not supported at the moment 
-Examples:  "ra" > 12345 and "color" != 'blue' or "band" IN (1,2,3)
+* grouping by parentheses is not supported at the moment
+* '${FILTER_SEP}' serves as groups separator at the moment 
+Examples:  
+  "ra" > 5 and "color" != 'blue' or "band" IN (1,2,3)
+  "ra" < 5 or "ra" > 6 _AND_ "dec" < -1 or "dec" > 1
 `;
 
 export const NULL_TOKEN = '%NULL';          // need to match DbAdapter.NULL_TOKEN
 
+// condition separator
 const COND_SEP = new RegExp('( and | or )', 'i');
-const FILTER_SEP = '_AND_';                 // internal use need to match what's defined in TableServerRequest.FILTER_SEP
+
+/**
+ * returns the number of individual filters
+ * @param filterInfoStr
+ * @returns {number}
+ */
+export function getNumFilters(filterInfoStr) {
+    if (filterInfoStr) {
+        return Math.round((filterInfoStr.split(new RegExp(`( and | or |${FILTER_SEP})`, 'i')).length+1)/2);
```

Comment:
Took me awhile to figure out why you needed the round and `/2`.  Not sure you meant to do this, but if you remove the capturing `()`, you can just return the length of the result.

`filterInfoStr.split(new RegExp(` and | or |${FILTER_SEP}`, 'i')).length`

Reply:
Thank you.